scripts/shell/build_node_source.sh

```diff
@@ -7,7 +7,8 @@ build_node(){
 	./configure --enable-static --without-intl
 	python3 patch_node_build.py -p ./
 	rm patch_node_build.py
-	make -j4
+	num_jobs=$(command -v nproc > /dev/null 2>&1 && nproc || echo 4)
+	make -j $num_jobs
```

Comment:
It used to be the following.
```sh
make -j `nproc`
```

The problem is the Github env sometimes is weird that `nproc` may return a very large number way greater than 4 though it only allocates 2-4 cores. That causes the whole docker build hang.

Now, the docker builds are suspended in the workflow. That's not a problem.

So, the question is what's the difference between this 2-line change and that 1-line code?

Reply:
I am not sure `build_node_source.sh` will be used on macOS or not. If it is only used on a Linux host or a docker container, we just need the `make -j $(nproc)`. But if it will be reused on macOS, `nproc` does not exist, so we need to fallback to a default value. That's why I use the `command -v nproc` checking.

Reply:
`nproc` works on mac, but the problem remains as `nproc` sometimes returns the VM host's CPU core count instead of the VM guest's. That's the concern I'm holding. I suggest a docker arg (defaults to 4) and you may pass whatever number in your build script. What do you think?

Reply:
I agree with using a docker arg.